### PR TITLE
add: `webstorm-bin`

### PR DESCRIPTION
**Pull request by jimed-rand**
Adding WebStorm as available for free (non-commercial usages) and part of requests I made based on #6744. The versions are 2024.3.1.

### CODE REVIEW

File: packages/webstorm-bin/webstorm-bin.pacscript [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+pkgdesc="Powerful and versatile IDE for JavaScript, TypeScript, HTML, CSS, React, Angular, Vue, and Node.js development."
+external_connection="true"
+source_amd64=("https://download.jetbrains.com/${gives}/WebStorm-${pkgver}.tar.gz")
+sha256sums_amd64=("f9122a02312bee9d06c77774cad37b32ac0dcbc460af4ac8b8059a1780d16018")
+source_arm64=("https://download.jetbrains.com/${gives}/WebStorm-${pkgver}-aarch64.tar.gz")
+sha256sums_arm64=("e1b34cf2456233f6a7aa079e7c2af23bc88d4b29bbddcf6d7a5b4e0432e38db3")
+repology=("project: webstorm")
+
+package() {
+  cd "${_archive}"
```
Comment (Elsie19) [inline packages/webstorm-bin/webstorm-bin.pacscript:15]: `${_archive}` is going to be deprecated, and we don't accept new PRs that contain it. Please change it to the unpacked download directory!

File: packages/webstorm-bin/webstorm-bin.pacscript [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+gives="webstorm"
+arch=("amd64" "arm64")
+maintainer=("James Ed Randson <jimedrand@disroot.org>")
+pkgdesc="Powerful and versatile IDE for JavaScript, TypeScript, HTML, CSS, React, Angular, Vue, and Node.js development."
+external_connection="true"
+source_amd64=("https://download.jetbrains.com/${gives}/WebStorm-${pkgver}.tar.gz")
+sha256sums_amd64=("f9122a02312bee9d06c77774cad37b32ac0dcbc460af4ac8b8059a1780d16018")
+source_arm64=("https://download.jetbrains.com/${gives}/WebStorm-${pkgver}-aarch64.tar.gz")
+sha256sums_arm64=("e1b34cf2456233f6a7aa079e7c2af23bc88d4b29bbddcf6d7a5b4e0432e38db3")
+repology=("project: webstorm")
```
Comment (Elsie19) [inline packages/webstorm-bin/webstorm-bin.pacscript:12]:
```suggestion
source_amd64=("https://download.jetbrains.com/${gives}/WebStorm-${pkgver}.tar.gz")
sha256sums_amd64=("f9122a02312bee9d06c77774cad37b32ac0dcbc460af4ac8b8059a1780d16018")
source_arm64=("https://download.jetbrains.com/${gives}/WebStorm-${pkgver}-aarch64.tar.gz")
sha256sums_arm64=("e1b34cf2456233f6a7aa079e7c2af23bc88d4b29bbddcf6d7a5b4e0432e38db3")
repology=("project: webstorm")
external_connection=true
```

Comment (Elsie19) [reply packages/webstorm-bin/webstorm-bin.pacscript:12]: Put it at the bottom and remove quotes (this is just our standard that we've assumed)

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: packages/webstorm-bin/webstorm-bin.pacscript (hunk 1 above)
Comment (jimed-rand) [reply packages/webstorm-bin/webstorm-bin.pacscript]: Got it, i've fix it with `cd "${srcdir}/${_archive}"`.

**Comment by jimed-rand** [comment]
It's ready to review, I've fixed and follow your suggestions @Elsie19.

File: packages/webstorm-bin/webstorm-bin.pacscript [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+external_connection=true
+
+package() {
+  cd "${srcdir}/WebStorm-243.22562.112" || {
+    echo "Directory not found"
+    exit 1
+  }
```
Comment (Elsie19) [inline packages/webstorm-bin/webstorm-bin.pacscript:18]: This should never ever fail and if it does there are bigger issues than that. Assume it works and remove the guard.

File: packages/webstorm-bin/webstorm-bin.pacscript [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  mkdir -p "${pkgdir}/usr/share/licenses/webstorm"
+  mkdir -p "${pkgdir}/usr/share/webstorm"
+  mkdir -p "${pkgdir}/usr/share/pixmaps"
+
+  for item in bin jbr lib plugins product-info.json license; do
+    if [ ! -e "$item" ]; then
+      echo "Error: $item not found"
+      exit 1
+    fi
```
Comment (Elsie19) [inline packages/webstorm-bin/webstorm-bin.pacscript:28]: Assume everything works. It always should.

File: packages/webstorm-bin/webstorm-bin.pacscript (hunk 3 above)
Comment (jimed-rand) [reply packages/webstorm-bin/webstorm-bin.pacscript:18]: It actually works, but it couldn't work in pre-commit side

Comment (Elsie19) [reply packages/webstorm-bin/webstorm-bin.pacscript:18]: Remove the guards please. I will deal with precommit.

Comment (jimed-rand) [reply packages/webstorm-bin/webstorm-bin.pacscript:18]: > Remove the guards please. I will deal with precommit.

Alright, I'll do it after exams tomorrow.

Comment (jimed-rand) [reply packages/webstorm-bin/webstorm-bin.pacscript:18]: Got it, it's fixed today. I have remove the guards. Sorry for waiting too long.

Comment (jimed-rand) [reply packages/webstorm-bin/webstorm-bin.pacscript:18]: Got it, it's fixed today. I have remove the guards. Sorry for waiting too long.

File: packages/webstorm-bin/webstorm-bin.pacscript (hunk 4 above)
Comment (jimed-rand) [reply packages/webstorm-bin/webstorm-bin.pacscript:28]: What should I give the changes for this ?

Comment (jimed-rand) [reply packages/webstorm-bin/webstorm-bin.pacscript]: Fine, I have remove that and now pre-commit are works and ready to merge.

File: packages/webstorm-bin/webstorm-bin.pacscript [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+gives="webstorm"
+arch=("amd64" "arm64")
+maintainer=("James Ed Randson <jimedrand@disroot.org>")
+pkgdesc="Powerful and versatile IDE for JavaScript, TypeScript, HTML, CSS, React, Angular, Vue, and Node.js development."
+source_amd64=("https://download.jetbrains.com/${gives}/WebStorm-${pkgver}.tar.gz")
+sha256sums_amd64=("f9122a02312bee9d06c77774cad37b32ac0dcbc460af4ac8b8059a1780d16018")
+source_arm64=("https://download.jetbrains.com/${gives}/WebStorm-${pkgver}-aarch64.tar.gz")
+sha256sums_arm64=("e1b34cf2456233f6a7aa079e7c2af23bc88d4b29bbddcf6d7a5b4e0432e38db3")
+repology=("project: webstorm")
+external_connection=true
```
Comment (Elsie19) [inline packages/webstorm-bin/webstorm-bin.pacscript:12]: This isn't needed based on your `package()` function.

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: packages/webstorm-bin/webstorm-bin.pacscript (hunk 5 above)
Comment (jimed-rand) [reply packages/webstorm-bin/webstorm-bin.pacscript:12]: Did you mean the `external_connection=true`? I'll delete it if unnecessary.

Comment (jimed-rand) [reply packages/webstorm-bin/webstorm-bin.pacscript:12]: Or for `aarch64/arm64` versions ?

Comment (Elsie19) [reply packages/webstorm-bin/webstorm-bin.pacscript:12]: external_connection.

File: packages/webstorm-bin/webstorm-bin.pacscript [hunk 6]
```diff
@@ -9,7 +9,6 @@ sha256sums_amd64=("f9122a02312bee9d06c77774cad37b32ac0dcbc460af4ac8b8059a1780d16
 source_arm64=("https://download.jetbrains.com/${gives}/WebStorm-${pkgver}-aarch64.tar.gz")
 sha256sums_arm64=("e1b34cf2456233f6a7aa079e7c2af23bc88d4b29bbddcf6d7a5b4e0432e38db3")
 repology=("project: webstorm")
```
Comment (jimed-rand) [inline packages/webstorm-bin/webstorm-bin.pacscript:11]: It's fixed now, you can review again or merge it if everything's okay.

**Pull request merged**